anvil/src/shell.rs

```diff
@@ -221,47 +211,58 @@ impl PointerGrab for ResizeSurfaceGrab {
 
     fn button(
         &mut self,
-        handle: &mut PointerInnerHandle<'_>,
-        button: u32,
-        state: ButtonState,
-        serial: Serial,
-        time: u32,
+        data: &mut AnvilState<BackendData>,
+        dh: &DisplayHandle,
+        handle: &mut PointerInnerHandle<'_, AnvilState<BackendData>>,
+        event: &ButtonEvent,
     ) {
-        handle.button(button, state, serial, time);
+        handle.button(event.button, event.state, event.serial, event.time);
         if handle.current_pressed().is_empty() {
             // No more buttons are pressed, release the grab.
-            handle.unset_grab(serial, time);
+            handle.unset_grab(event.serial, event.time);
 
             // If toplevel is dead, we can't resize it, so we return early.
-            if !self.window.toplevel().alive() | self.window.toplevel().wl_surface().is_none() {
+            if !self.window.toplevel().alive() {
                 return;
             }
 
             #[cfg_attr(not(feature = "xwayland"), allow(irrefutable_let_patterns))]
             if let SurfaceKind::Xdg(xdg) = &self.window.toplevel() {
-                let ret = xdg.with_pending_state(|state| {
+                xdg.with_pending_state(|state| {
                     state.states.unset(xdg_toplevel::State::Resizing);
                     state.size = Some(self.last_window_size);
                 });
-                if ret.is_ok() {
-                    xdg.send_configure();
+                xdg.send_configure();
+                if self.edges.intersects(ResizeEdge::TOP_LEFT) {
+                    let geometry = self.window.geometry();
+                    let mut location = data.space.window_location(&self.window).unwrap();
+
+                    if self.edges.intersects(ResizeEdge::LEFT) {
+                        location.x =
+                            self.initial_window_location.x + (self.initial_window_size.w - geometry.size.w);
+                    }
+                    if self.edges.intersects(ResizeEdge::TOP) {
+                        location.y =
+                            self.initial_window_location.y + (self.initial_window_size.h - geometry.size.h);
+                    }
+
+                    data.space.map_window(&self.window, location, true);
                 }
 
-                with_states(self.window.toplevel().wl_surface().unwrap(), |states| {
+                with_states(self.window.toplevel().wl_surface(), |states| {
                     let mut data = states
                         .data_map
                         .get::<RefCell<SurfaceData>>()
                         .unwrap()
                         .borrow_mut();
                     if let ResizeState::Resizing(resize_data) = data.resize_state {
-                        data.resize_state = ResizeState::WaitingForFinalAck(resize_data, serial);
+                        data.resize_state = ResizeState::WaitingForFinalAck(resize_data, event.serial);
```

Comment:
Once #598 is merged, send_configure() serial should be used here, instead of unrelated button serial. Ack serial is the serial that was used to send the configure. As protocol description of `ack_configure` states `serial - is the serial from the configure event`

Reply:
Let's leave this for another PR, as it's not related, and requires more investigation as to why are we waiting for that ack to begin with.